### PR TITLE
Add test workflows via GitHub actions

### PR DESCRIPTION
**Pull request by auria**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    name: Integration tests
+    runs-on: self-hosted
+    needs: lint-and-unit-tests
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
```
Comment (r-f-g) [inline .github/workflows/tests.yml:44]: I think you should use `actions-operator` to be sure that juju and tox are installed. At the same time it provides the possibility to use GitHub secrets for the credentials and not directly.

**Review by r-f-g — COMMENTED** [review]
only one comment

File: .github/workflows/tests.yml (hunk 1 above)
Comment (auria) [reply .github/workflows/tests.yml:44]: `actions-operator` is not in the marketplace, and I'm trying to limit which actions can be set up by limiting to those created by GitHub or verified within the marketplace.

OTOH, the secrets I created in the main repository are not available to forked repositories that create a PR against the main repo. This is the reason why I switched to sourcing a file with credentials that are not directly usable for anybody outside of this test env. This together with the limitation that no tests will be run for new contributors until a maintainer approves it should reduce the risks.

**Review by r-f-g — APPROVED** [review]
LGTM

**Pull request merged**